modules/sdk-core/src/bitgo/utils/tss/baseTSSUtils.ts

```diff
@@ -163,6 +165,40 @@ export default class BaseTssUtils<KeyShare> extends MpcUtils implements ITssUtil
     return unsignedTx;
   }
 
+  /**
+   * Create a tx request from params for message signing
+   *
+   * @param params
+   * @param apiVersion
+   * @param preview
+   */
+  async createTxRequestWithIntentForMessageSigning(
+    params: IntentOptionsBase,
+    apiVersion: TxRequestVersion = 'full',
+    preview?: boolean
+  ): Promise<TxRequest> {
+    const intentOptions: PopulatedIntentBase = {
```

Comment:
u need an optional parameter for `custodianMessageId` to support message requests initiated from MMI. This will be stored in the txRequest model and used to send updates back to the extension.

Reply:
Is that needed for all txrequest or just message signing?

Reply:
just message signing -> I have created two new intents to be used in the txRequest API

There is a different field for txs -> `custodianTransactionId`

Reply:
https://github.com/BitGo/bitgo-microservices/pull/24139

Reply:
can we handle this in a follow up PR?